sync/trypush.py

```diff
@@ -433,46 +434,39 @@ def wpt_tasks(self, force_update=False):
         :return: List of tasks
         """
         if not force_update and "tasks" in self._data:
-            return self._data["tasks"]
-
-        try:
-            wpt_tasks = tc.get_wpt_tasks(self.taskgroup_id)
-        except ValueError:
-            # If this happens we may have the wrong taskgroup id
+            tasks = tc.TaskGroup(self.taskgroup_id, self._data["tasks"])
+        else:
             task_id = tc.normalize_task_id(self.taskgroup_id)
             if task_id != self.taskgroup_id:
                 self.taskgroup_id = task_id
-            wpt_tasks = tc.get_wpt_tasks(self.taskgroup_id)
+
+            tasks = tc.TaskGroup(self.taskgroup_id)
+            tasks.refresh()
+
+        self._data["tasks"] = tasks.tasks
+        return tasks
+
+    def wpt_tasks(self, force_update=False):
+        tasks = self.tasks(force_update)
+        wpt_tasks = tasks.view(tc.is_suite_fn("web-platform-tests"))
+        return wpt_tasks
+
+    def validate_tasks(self):
```

Comment:
Previously, tasks were "validated" with every call to `wpt_tasks`. Now, a bunch of points at which we may have identified `infra_fail` are no longer in place (I see only one call to `validate_tasks`), so I'm concerned we'll miss an infra failure or label it much later in the sync process. Validating the tasks in `try_push_complete` makes sense, but have you looked at all the old calls to `wpt_tasks` to make sure we're not counting on updating the `infra_fail` flag at those points?

Reply:
I think it makes sense that we only validate in the entry point. We might be missing some validation in the landing code, but I think we want different semantics in that case anyway.